### ISSUE TITLE
Support multiple calls to Push-OutputBinding to same `-Name`

### ISSUE THREAD

**Issue by SteveL-MSFT** [Azure/azure-functions-powershell-worker#152]
Currently, there is a limitation that if you want to push two items to the same storage queue, you must call `Push-OutputBinding` only once with an array of those items.  It's more natural in the script to call `Push-OutputBinding` multiple times at the time of creation of that queue item rather than collecting them and then pushing at the end.

**Comment by devblackops** [comment]
This limitation is also present when working with table storage. Would this be an issue with **ALL** the output bindings?

**Comment by TylerLeonhardt** [comment]
@devblackops yeah it kinda applies to all... except `http` out binding.

I think the question here is... What's the behavior for arrays in output bindings?

Today you can:

```pwsh
$arr = @()

$arr += "foo"
$arr += "bar"

Push-OutputBinding MyArray $arr
```

which, if MyArray goes to a Queue, it will add X items to the Queue

But what you can't do is:

```
Push-OutputBinding MyArray "foo"
Push-OutputBinding MyArray "bar"
```

and have it keep an array of things you want to send out.

Maybe we need like a `-Append` switch that will basically keep an array inside the output binding?

**Comment by devblackops** [comment]
I suppose there are legit use cases when I would like to return multiple items to the same output binding, with some of them being arrays and not others. It would be natural to allow calling `Push-OutputBinding` multiple times throughout the function without having to build up an array to provide at the end.

Something like:

```powershell
Push-OutputBinding -Name foo -Value 'bar'
Push-OutputBinding -Name foo -Value @(42, 'asdf')
Push-OutputBinding -Name foo -Value 12345
```

Would it make sense to store the values in `$script:_OutputBindings[$Name]` as an array by default with it normally being a count of 1?

The example above would have `$script:_OutputBindings['foo']` looking like:

```powershell
@(
  'bar'
  @(
    42
    'asdf'
  )
  12345
)
```

**Comment by SteveL-MSFT** [comment]
My preference is by default, multiple calls will queue up the objects to be pushed at the end (since this appears to be a limitation with Functions).  We can optionally have a `-Clobber` switch if needed.  Building the array to push at the end is a workaround and requires everyone to have that code.

**Comment by TylerLeonhardt** [comment]
I suppose something with a verb of `Push` makes sense for what you're saying too.

**Comment by daxian-dbw** [comment]
>  it kinda applies to all... except http out binding.

Does it really apply to all out bindings? What about blob?

I think one way to solve this is to have the worker share the output binding information with the module, so within `Push-OutputBinding`, it knows what type of binding the `Name` is, and then behave differently based on that information. For example, if the `Name` is actually a queue or table output, then `Push-OutputBinding` allows repeatedly calls to accumulate output values in a collection. But if the output binding is not one that support a collection of output, then `Push-OutputBinding` will fail with the proper error message.

**Comment by TylerLeonhardt** [comment]
I mentioned in Dongbo's PR... I'd like to get more customer feedback on this... It could be a bit too magic-y that certain out bindings behave differently with `Push-OutputBinding`

**Comment by daxian-dbw** [comment]
> It could be a bit too magic-y that certain out bindings behave differently with Push-OutputBinding

I thought about it when mentioning this possible UX enhancement in Teams, and believe that the behavior can be well documented.
The behavior changes based on the output binding you are targeting when calling `Push-OutputBinding` -- the argument for `-Name` explicitly indicates the output binding. When incorrect usage is caught, the error message can be made very clear why you cannot do it with that type of output binding with the URL to the document.

**Comment by daxian-dbw** [comment]
A not very accurate analogy: it's like the relationship between provider cmdlets such as `Set-Item` and various providers.
`Set-Item` might have different behavior depending on the argument to `-Path`, which explicitly indicates the target provider.

In our case, different output binding types are the providers, and `Push-OutputBinding` is `Set-Item`.

**Comment by TylerLeonhardt** [comment]
I see what you're saying... 

Things that should throw:
* http
* blob?

Things that should add to an array:
* queue
* event grid
* event hub
* cosmos db
* table
* service bus
* all the Microsoft graph ones

There's [just so many out bindings](https://docs.microsoft.com/en-us/azure/azure-functions/functions-triggers-bindings#supported-bindings)... we'd need to decide which does which and what the default behavior will be in the event that a new binding is added (like SignalR which was added last week).

**Comment by daxian-dbw** [comment]
> we'd need to decide which does which

Agreed. We will have to play with them ourselves first anyways.

> what the default behavior will be in the event that a new binding is added

I personal opinion is: start conservatively -- assuming no collection support for new out bindings.

**Comment by joeyaiello** [comment]
Given that no output bindings actually support this output yet (except HTTP, but that requires a wrapper object):

```
@(
  'bar'
  @(
    42
    'asdf'
  )
  12345
)
```

I think this should probably work such that line 2 unrolls the items and appends them to the binding (for bindings like queues where collections are supported). 

```powershell
Push-OutputBinding -Name foo -Value 'bar'
Push-OutputBinding -Name foo -Value @(42, 'asdf')
Push-OutputBinding -Name foo -Value 12345
```

If collections aren't supported, we should continue to throw an error and suggest that the user try `-Force` to overwrite the value.

We also had some discussion about whether we should change this to `-Clobber` given the overloading of `-Force`.

We also talked about whether HTTP might have a "simple" mode that only takes values for the body. This would be the only yet known scenario where @devblackops output above might make sense.

**Comment by daxian-dbw** [comment]
Fixed via #168
